unordered_associative_task_2/student/src/impl.cpp

```diff
@@ -41,9 +72,55 @@ filtering_map groupDuplicates (const std::vector<std::string>& dataSource, Filte
     return grouped;
 }
 
-std::vector<std::vector<std::string> > findDuplicates(const std::string &rootPath)
+/** @todo With help of std filesystem, lists all regular files under specified directory recursively*/
+std::vector<std::string> listFiles(const std::string& directory)
+{
+    std::vector<std::string> fileNames;
+    for (auto& p : fs::recursive_directory_iterator(directory)) {
+        if (fs::is_regular_file(p)) {
+            fileNames.push_back(p.path().string());
+        }
+    }
+    return fileNames;
+}
+
+/** @todo Implement function that will remove group in if it has only one or zero elements */
+void removeUniqueGroups(filtering_map& filteredData)
+{
+    for (auto it = filteredData.begin(); it != filteredData.end();) {
+        if (it->second.size() == 1)
+            it = filteredData.erase(it);
+        else
+            ++it;
+    }
+}
+
+/** @todo Implement function that will transform map to a vector */
+std::vector<std::vector<std::string>> flattenGrouped(const filtering_map& grouped)
+{
+    std::vector<std::vector<std::string>> groups;
+    for (auto const& group : grouped) {
+        if (!group.second.empty()) {
+            std::vector<std::string> files(group.second.begin(), group.second.end());
+            groups.emplace_back(files);
```

Comment:
in this case, you should pass not the _files_, but _group.second.begin(), group.second.end()_
**You should pass initialization arguments for object to emplace function! Not the lvalue or rvalue!**